File: 234. Palindrome Linked List/isPalindrome.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
    var r_list = []
    while (head != null) {
        r_list.push(head.val)
        head = head.next
    }
    re_list = [...r_list].reverse()
    for (let i = 0; i < r_list.length; i++) {
        if (r_list[i] !== re_list[i]) {
            return  false
        }
    }
    return true
};

// Runtime 234 ms / Memory 88 MB
// debug: https://jsfiddle.net/

File: 234. Palindrome Linked List/isPalindrome.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        // create list
        vector<int> values;
        ListNode *node = head;
        while (node) {
            values.push_back(node->val);
            node = node->next;
        }
        // check palindrome
        vector<int> sortValues = values;
        reverse(sortValues.begin(),sortValues.end());
        return (sortValues==values);
    }
};